### PR TITLE
[torch-mlir]enable const op ut

### PR DESCRIPTION
**Pull request by Yancey1989**

### CODE REVIEW

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp [hunk 1]
```diff
@@ -1519,11 +1519,18 @@ LogicalResult ConvertAtenOp<ValueTensorLiteralOp>::matchAndRewrite(
     ValueTensorLiteralOp op,
     OpAdaptor adaptor,
     ConversionPatternRewriter& rewriter) const {
-  auto outputTy = getTypeConverter()
-                      ->convertType(op.getType())
-                      .template cast<RankedTensorType>();
-  rewriter.replaceOpWithNewOp<mhlo::ConstOp>(op, outputTy, adaptor.value());
-
+  MLIRContext* context = op->getContext();
+  if (auto elements = op.valueAttr().dyn_cast<DenseIntElementsAttr>()) {
```
Comment (wyzero) [inline pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:1523]: Why do we need to treat int attribute specially?

Comment (Yancey1989) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:1523]: It seems `mhlo.constant` does not support `sint` type, so I should deal with INT especially.

``` text
error: 'mhlo.constant' op result #0 must be statically shaped tensor of floating-point or pred (AKA boolean or 1-bit integer) or 8/16/32/64-bit signless integer or 8/16/32/64-bit unsigned integer or complex type with 32-bit float or 64-bit float elements values, but got 'tensor<4xsi64>'
// -----// IR Dump After ConvertTorchToMhlo Failed //----- //
"func.func"() ({
  %0 = "mhlo.constant"() {value = dense<[1, 2, 3, 4]> : tensor<4xsi64>} : () -> tensor<4xsi64>
  %1 = "torch_c.from_builtin_tensor"(%0) : (tensor<4xsi64>) -> !torch.vtensor<[4],si64>
  "func.return"(%1) : (!torch.vtensor<[4],si64>) -> ()
}) {arg_attrs = [], function_type = () -> !torch.vtensor<[4],si64>, sym_name = "main"} : () -> ()
```

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp [hunk 2]
```diff
@@ -1519,11 +1519,18 @@ LogicalResult ConvertAtenOp<ValueTensorLiteralOp>::matchAndRewrite(
...
+    Type elemTy = op.valueAttr().getElementType();
+    unsigned bitWidth = elemTy.getIntOrFloatBitWidth();
+    Type builtinTensorElemTy = IntegerType::get(context, bitWidth);
+    rewriter.replaceOpWithNewOp<mhlo::ConstOp>(
+        op, elements.mapValues(builtinTensorElemTy, [&](const APInt& v) {
+          return APInt(bitWidth, v.getSExtValue());
```
[the remark this answers is not in the document]
Comment (Yancey1989) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:1529]: Will use `rewriter.replaceOpWithNewOp<mhlo::ConstOp>(op, op.valueAttr());` goes well without any special treating.

File: pytorch_blade/tests/disc/ops/test_const.py [hunk 3]
```diff
@@ -49,19 +50,23 @@ def return_const_scalar():
...
-                %1 = "std.constant"() {value = 5 : i64} : () -> i64
-                # CHECK: mhlo.constant dense<5> : tensor<i64>
-                %2 = "xla_hlo.constant"() {value = dense<5> : tensor<i64>} : () -> tensor<i64>
-                "std.return"(%2) : (tensor<i64>) -> () loc(unknown)
-              } loc(unknown)
-            } loc(unknown)
-        """
-        FileCheck().run(expect_str, disc_bytes)
+        actual_str, _, _, _ = mlir.cvt_torchscript_to_mhlo(graph)
+        if read_bool_from_env("TORCH_DISC_USE_TORCH_MLIR", False):
```
[the remark this answers is not in the document]
Comment (Yancey1989) [reply pytorch_blade/tests/disc/ops/test_const.py]: done

**Review by fortianyou — APPROVED** [review]
LGTM

**Pull request merged**